clientconn.go

```diff
@@ -95,6 +96,14 @@ func WithTimeout(d time.Duration) DialOption {
 	}
 }
 
+// WithNetwork returns a DialOption that specifies the network on which
+// the connection will be established.
+func WithNetwork(network string) DialOption {
```

Comment:
This feels too specific. It doesn't help this support any other address types either. Why don't we simply accept a dialer option that takes complete responsibility for turning the address into a net.Conn?